File: 0268-missing-number/0268-missing-number.java
class Solution {
    static class Pair{
        int val;
        int idx;
        Pair(int val,int idx){
            this.val=val;
            this.idx=idx;
        }
    }
    public int missingNumber(int[] arr) {
        Arrays.sort(arr);
        ArrayList<Pair>list=new ArrayList<>();
        for(int i=0;i<arr.length;i++){
            list.add(new Pair(arr[i],i));
        }
        for(int i=0;i<list.size();i++){
            if(list.get(i).val!=list.get(i).idx) return i;
        }
        return arr.length;
    }
}

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    class Pair {
    public:
        int val;
        int idx;

        Pair(int val, int idx) {
            this->val = val;
            this->idx = idx;
        }
    };

    int missingNumber(vector<int>& v) {
        sort(v.begin(), v.end());

        vector<Pair> list;

        for (int i = 0; i < v.size(); i++) {
            list.emplace_back(v[i], i);
        }

        for (int i = 0; i < list.size(); i++) {
            if (list[i].val != list[i].idx)
                return list[i].idx;
        }

        return v.size();
    }
};